File: 210/level1/숫자문자열과영단어.java
// 20220815
package level1;

class Main {
    public static void main(String[] args) throws Exception {
        String s = "one4seveneight";

        Solution sol = new Solution();
        System.out.println("result : " + sol.solution(s));
    }
}


class Solution {
    public int solution(String s) {
        String arr[]={"zero","one","two","three","four","five","six","seven","eight","nine"};
        for(int i=0;i<10;i++){
            s=s.replace(arr[i],Integer.toString(i));
        }
        return Integer.parseInt(s);
    }
}

File: JunJangE/level1/숫자문자열과영단어.kt
package level1



fun main() {
    println(Solution().solution("one4seveneight"))
    
}

class Solution {

    fun solution(s: String): Int {
        val map = hashMapOf<String, String>("zero" to "0", "one" to "1", "two" to "2",  "three" to "3", "four" to "4", "five" to "5", "six" to "6",
                "seven" to "7", "eight" to "8", "nine" to "9")

        var answer: String = s

        for ((key, value) in map){
            answer = answer.replace(key, value)

        }
        return answer.toInt()
    }
}
